src/api/entrypoint.ts

```diff
@@ -0,0 +1,88 @@
+import joi from 'joi';
+import { promises as fs } from 'fs';
+import path from 'path';
+import { scopedLogger } from '@logger';
+
+const log = scopedLogger('runner');
+
+const actionPayloadSchema = joi
+  .object({
+    action: joi.string().required(),
+    data: joi.object(),
+  })
+  .strict();
+
+interface Schema {
+  action: string;
+  schema: joi.ObjectSchema;
+  run: (data: any) => any;
+}
+const schemas: Record<string, Schema> = {};
+
+export async function setupApi() {
+  const files = await fs.readdir(path.join(__dirname, './schemas'), {
+    withFileTypes: true,
+  });
+  const fileNames = files
+    .filter((v) => {
+      // js files are supported in production use, where ts are compiled to js
+      const isCodefile = v.name.endsWith('.js') || v.name.endsWith('.ts');
+      const isNormalFile = v.isFile();
+      return isCodefile && isNormalFile;
+    })
+    .map((v) => v.name);
+
+  const importedData = fileNames.map((name) =>
+    require(path.join(__dirname, './schemas', name)),
+  );
+  importedData.forEach((data: Schema) => {
+    schemas[data.action] = data;
+  });
+}
+
+/*
+ ** main entrypoint for sandboxed methods to be called
+ ** all parameters of this method are untrusted
+ */
+export async function callApiMethod(payload: any): Promise<any> {
```

Comment:
ahhh yes, all 1 parameter is untrusted. Is this going to serialize all data into an object, or is this supposed to accept multiple parameters?

Reply:
updated comment